File: PetCareManager/src/PetCareManager.Core/Model/VaccinationAggregate/Vaccination.cs
// Licensed to the .NET Foundation under one or more agreements.
// The .NET Foundation licenses this file to you under the MIT license.

namespace PetCareManager.Core;

public class Vaccination
{
    public Guid VaccinationId { get; set; }

    /// <summary>
    /// Gets or sets the tenant ID for multi-tenant isolation.
    /// </summary>
    public Guid TenantId { get; set; }
    public Guid PetId { get; set; }
    public Pet? Pet { get; set; }
    public string Name { get; set; } = string.Empty;
    public DateTime DateAdministered { get; set; }
    public DateTime? NextDueDate { get; set; }
    public string? VetName { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: PetCareManager/src/PetCareManager.WebApp/projects/pet-care-manager/src/app/models/vaccination.ts
export interface Vaccination {
  vaccinationId: string;
  petId: string;
  name: string;
  dateAdministered: string;
  nextDueDate?: string;
  vetName?: string;
  createdAt: string;
}

export interface CreateVaccinationDto {
  petId: string;
  name: string;
  dateAdministered: string;
  nextDueDate?: string;
  vetName?: string;
}

export interface UpdateVaccinationDto {
  vaccinationId: string;
  petId: string;
  name: string;
  dateAdministered: string;
  nextDueDate?: string;
  vetName?: string;
}
